optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/view/IndictmentView.java

```diff
@@ -0,0 +1,87 @@
+package org.optaplanner.openshift.employeerostering.shared.roster.view;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.score.constraint.ConstraintMatch;
+import org.optaplanner.core.api.score.constraint.Indictment;
+import org.optaplanner.persistence.jackson.api.score.ScoreJacksonJsonSerializer;
+import org.optaplanner.persistence.jackson.api.score.buildin.hardsoft.HardSoftScoreJacksonJsonDeserializer;
+
+public class IndictmentView {
+
+    private List<Rules> ruleMatchedList;
+    private List<HardSoftScore> scoreList;
+
+    public IndictmentView() {
+
+    }
+
+    public IndictmentView(Indictment indictment) {
+        ruleMatchedList = new ArrayList<>(indictment.getConstraintMatchCount());
+        scoreList = new ArrayList<>(indictment.getConstraintMatchCount());
+
+        for (ConstraintMatch cm : indictment.getConstraintMatchSet()) {
+            ruleMatchedList.add(Rules.forConstraintMatch(cm));
+            scoreList.add((HardSoftScore) cm.getScore());
+        }
+    }
+
+    public IndictmentView(List<Rules> ruleMatched, List<HardSoftScore> score) {
+        this.ruleMatchedList = ruleMatched;
+        this.scoreList = score;
+    }
+
+    public List<Rules> getRuleMatchedList() {
+        return ruleMatchedList;
+    }
+
+    @JsonSerialize(contentUsing = ScoreJacksonJsonSerializer.class)
+    @JsonDeserialize(contentUsing = HardSoftScoreJacksonJsonDeserializer.class)
+    public List<HardSoftScore> getScoreList() {
+        return scoreList;
+    }
+
+    public void setRuleMatchedList(List<Rules> ruleList) {
+        this.ruleMatchedList = ruleList;
+    }
+
+    public void setScoreList(List<HardSoftScore> scoreList) {
+        this.scoreList = scoreList;
+    }
+
+    public static enum Rules {
+        MISSING_REQUIRED_SKILL,
+        EMPLOYEE_ALREADY_HAS_A_SHIFT_ON_DATE,
+        EMPLOYEE_HAS_SHIFTS_LESS_THAN_1O_HOURS_APART,
+        UNDESIRED_TIMESLOT_FOR_EMPLOYEE,
+        DESIRED_TIMESLOT_FOR_EMPLOYEE,
+        UNAVALIABLE_TIMESLOT_FOR_EMPLOYEE,
+        EMPLOYEE_IS_NOT_ROTATION_EMPLOYEE,
+        UNKNOWN;
+
+        public static Rules forConstraintMatch(ConstraintMatch cm) {
+            switch (cm.getConstraintName()) {
```

Comment:
In theory, matching the constraintName isn't enough. A constraintMatchTotal is unique by the combination of constraintPackage and constraintName.
Ideally, with the code trick above, each Constraint enum has a 2 String fields: String constraintPackage and String constraintName. Both are being checked to be equal in that for loop.